node/impl/full/eth.go

```diff
@@ -461,22 +461,37 @@ func (a *EthModule) EthGetTransactionHashByCid(ctx context.Context, cid cid.Cid)
 
 func (a *EthModule) EthGetTransactionCount(ctx context.Context, sender ethtypes.EthAddress, blkParam ethtypes.EthBlockNumberOrHash) (ethtypes.EthUint64, error) {
 	addr, err := sender.ToFilecoinAddress()
+
 	if err != nil {
-		return ethtypes.EthUint64(0), nil
+		return ethtypes.EthUint64(0), xerrors.Errorf("invalid address: %w", err)
```

Comment:
this will need to be checked against eth implementation behaviour - if you supply an invalid address, will it error or return nothing?

Reply:
If an invalid address is provided (e.g., wrong length, non-hexadecimal characters, etc), eth returns an error.

Reply:
ACK, and it looks like the other methods in here that do this conversion also error here, so 👍